Tests/test_SeqIO_index.py

```diff
@@ -46,15 +45,11 @@ def add_prefix(key):
 
 
 def gzip_open(filename, format):
-    # At time of writing, under Python 3.2.2 seems gzip.open(filename, mode)
-    # insists on giving byte strings (i.e. binary mode)
-    # See http://bugs.python.org/issue13989
-    if sys.version_info[0] < 3 or format in SeqIO._BinaryFormats:
-        return gzip.open(filename)
-    handle = gzip.open(filename)
-    data = handle.read()  # bytes!
-    handle.close()
-    return StringIO(data.decode())
+    """Open gzip file in either binary or text mode as needed."""
+    if format in SeqIO._BinaryFormats:
```

Comment:
Some day we should try and get rid of `SeqIO._BinaryFormats`

Reply:
Yes, and this pull request is a small step in that direction.